src/TimeManager.jl

```diff
@@ -59,123 +50,82 @@ datetime_to_strdate(datetime::Dates.DateTime) =
     string(lpad(Dates.day(datetime), 2, "0"))
 
 """
-    AbstractFrequency
+    time_to_period(s::String)
 
-This is an abstract type for the frequency of the callback function.
-"""
-abstract type AbstractFrequency end
-
-"""
-    Monthly
-A concrete type for the monthly frequency of the callback function.
-"""
-struct Monthly <: AbstractFrequency end
+Convert a string to a `Dates.Period` object.
 
-"""
-    EveryTimestep
-A concrete type for the every-timestep frequency of the callback function.
-"""
-struct EveryTimestep <: AbstractFrequency end
-
-"""
-    trigger_callback(cs, ::Monthly)
+The string can be in one of two formats:
 
-Returns `true` if the current date is equal to or exceeds the saved first of the month at time of 00:00:00.
+1. **Months:** `<NUM>months`, where `<NUM>` is an integer representing the number of months. For example, `"2months"` represents 2 months.
+2. **Other units:** `<NUM><unit>`, where `<NUM>` is a number (integer or floating point) and `<unit>` is one of `secs`, `mins`, `hours`, or `days`.
```

Comment:
Isn't months just a special case of the generalized "other units"? Why does it have to be handled differently?

Reply:
All the other units have a fixed duration, but the number of seconds in month depends on the start date. But this doesn't really matter from a user point of view, so I will remove this distinction